monitors/BalanceMonitor.js

```diff
@@ -1,36 +1,63 @@
 // This module is used to monitor a list of addresses and their associated collateral, synthetic and ether balances.
 
 const { createFormatFunction, createEtherscanLinkMarkdown } = require("../common/FormattingUtils");
+const { createObjectFromDefaultProps } = require("../common/ObjectUtils");
 
 class BalanceMonitor {
   /**
    * @param {Object} logger Winston module used to send logs.
    * @param {Object} tokenBalanceClient Client used to query token balances for monitored bots and wallets.
    * @param tokenBalanceClient Instance of the TokenBalanceClient from the `financial-templates lib.
    * which provides synchronous access to address balances for a given expiring multi party contract.
-   * @param {List} botsToMonitor Array of bot objects to monitor. Each bot's `botName` `address`, `CollateralThreshold`
+   * @param {Object} config Object containing configuration for the balance monitor. Only option is a `botsToMonitor` 
+   * which is defines an array of bot objects to monitor. Each bot's `botName` `address`, `CollateralThreshold`
    *      and`syntheticThreshold` must be given. Example:
-   *      [{ name: "Liquidator Bot",
+   *      { botsToMonitor:[{ name: "Liquidator Bot",
    *         address: "0x12345"
    *         collateralThreshold: x1,
    *         syntheticThreshold: x2,
    *         etherThreshold: x3 },
-   *      ..]
+   *      ..] }
    * @param {Object} empProps Configuration object used to inform logs of key EMP information. Example:
    *      { collateralCurrencySymbol: "DAI",
             syntheticCurrencySymbol:"ETHBTC",
             priceIdentifier: "ETH/BTC",
             networkId:1 }
    */
-  constructor(logger, tokenBalanceClient, botsToMonitor, empProps) {
+  constructor(logger, tokenBalanceClient, config, empProps) {
     this.logger = logger;
 
     // Instance of the tokenBalanceClient to read account balances from last change update.
     this.client = tokenBalanceClient;
     this.web3 = this.client.web3;
 
-    // Bot addresses and thresholds to monitor.
-    this.botsToMonitor = botsToMonitor;
+    // Bot addresses and thresholds to monitor. If none provided then defaults to monitoring nothing. Configuration
```

Comment:
Below is the validation performed on the `botsToMonitor` structure ensuring that it is correctly named and formatted.